units/host_status.go

```diff
@@ -168,6 +164,17 @@ func (j *cloudHostReadyJob) setCloudHostStatus(ctx context.Context, m cloud.Mana
 			"hostStatus": hostStatus.String(),
 		})
 		return errors.Wrap(m.TerminateInstance(ctx, &h, evergreen.User, "cloud provider reported host failed to start"), "error terminating instance")
```

Comment:
should this also use the `NewHostTerminationJob` instead of terminating the host directly? (given starting hosts can run tasks now)

Reply:
I was thinking about this, but I also hesitate to change an existing implementation for an unrelated issue because I don't understand the original motivation behind the handling for failed/terminated hosts.

I do think the stopped host handling is probably more correct at least for the StatusFailed case (unsure about the StatusTerminated case because the host is already terminated in AWS), so I merged all of these cases so they all run the host termination job.